features/ccx-upgrades-data-eng/caching.feature

```diff
@@ -0,0 +1,220 @@
+      | INFERENCE_URL               | http://localhost:9090                  |
+      | SSO_ISSUER                  | http://localhost:8081/default |
+      | ALLOW_INSECURE              | 1                                      |
+      | RHOBS_URL                   | http://localhost:9091                  |
+      | OAUTHLIB_INSECURE_TRANSPORT | 1                                      |
+      | CACHE_ENABLED               | 1                                      |
+      | CACHE_TTL                   | 60                                     |
+      | CACHE_SIZE                  | 3                                      |
+    When I request the cluster endpoint in localhost:8000 with path aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/upgrade-risks-prediction
+    And I store the response for aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee for comparison
+    Then The status code of the response is 200
+    When I request the cluster endpoint in localhost:8000 with path aaaaaaaa-bbbb-cccc-dddd-000000000000/upgrade-risks-prediction
+    And I store the response for aaaaaaaa-bbbb-cccc-dddd-000000000000 for comparison
+    Then The status code of the response is 200
+    When I request the cluster endpoint in localhost:8000 with path 44444444-3333-2222-1111-111111111111/upgrade-risks-prediction
+    And I store the response for 44444444-3333-2222-1111-111111111111 for comparison
+    Then The status code of the response is 200
+    When I request the cluster endpoint in localhost:8000 with path 00000000-1111-2222-3333-444444444444/upgrade-risks-prediction
+    And I store the response for 00000000-1111-2222-3333-444444444444 for comparison
+    Then The status code of the response is 404
+    When I stop the mock CCX Inference Service
+    And I stop the mock RHOBS Service
+    And I request the cluster endpoint in localhost:8000 with path aaaaaaaa-bbbb-cccc-dddd-000000000000/upgrade-risks-prediction
+    Then The status code of the response is 200
+    And The response is identical to the previous response for aaaaaaaa-bbbb-cccc-dddd-000000000000
+    When I request the cluster endpoint in localhost:8000 with path 44444444-3333-2222-1111-111111111111/upgrade-risks-prediction
+    Then The status code of the response is 200
+    And The response is identical to the previous response for 44444444-3333-2222-1111-111111111111
+    When I request the cluster endpoint in localhost:8000 with path 00000000-1111-2222-3333-444444444444/upgrade-risks-prediction
+    Then The status code of the response is 404
+    And The response is identical to the previous response for 00000000-1111-2222-3333-444444444444
+    # Item for aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee is not in the cache anymore, so this fails as RHOBS / inference are not available
+    When I request the cluster endpoint in localhost:8000 with path aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/upgrade-risks-prediction
+    Then The status code of the response is 500
```

Comment:
it should be 503 IIRC

Reply:
or maybe the 500 is mapped to 503 in smart proxy

Reply:
probably. In the upgrades-end service itself, it's 500